### PR TITLE
fix: add organizationID query for user v2 ListUsers and clean up depeprecated attribute

### PR DESCRIPTION
**Pull request by stebenz**
Add organizationID as query for ListUsers and clean up the deprecated Organisation attributes in other queries.

This PR removes the following fields from API requests (user service v2):
- `organisation` from AddHumanUser (deprecated some time ago, `organization` still exists)
- `organization` from GetUserByID

### Definition of Ready

- [x] I am happy with the code
- [x] Short description of the feature/issue is added in the pr description
- [x] PR is linked to the corresponding user story
- [x] Acceptance criteria are met
- [x] All open todos and follow ups are defined in a new ticket and justified
- [x] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [x] No debug or dead code
- [x] My code has no repetitions
- [x] Critical parts are tested automatically
- [x] Where possible E2E tests are implemented
- [x] Documentation/examples are up-to-date
- [x] All non-functional requirements are met
- [ ] Functionality of the acceptance criteria is checked manually on the dev system.

### CODE REVIEW

File: proto/zitadel/user/v2beta/query.proto [hunk 1]
```diff
@@ -29,6 +29,7 @@ message SearchQuery {
         AndQuery and_query = 12;
         NotQuery not_query = 13;
         InUserEmailsQuery in_user_emails_query = 14;
+        ResourceOwnerQuery resource_owner_query = 15;
```
Comment (livio-a) [inline proto/zitadel/user/v2beta/query.proto:32]: naming question: resource_owner or organization?

Comment (livio-a) [reply proto/zitadel/user/v2beta/query.proto:32]: e.g. in v3 we designed an organizationIdQuery: https://docs-git-user-v2-list-zitadel.vercel.app/docs/apis/resources/user_service_v3/user-service-list-users

File: proto/zitadel/user/v2beta/query.proto [hunk 2]
```diff
@@ -200,6 +201,16 @@ message InUserEmailsQuery {
...
 
+//ResourceOwnerQuery always equals
+message ResourceOwnerQuery {
+    string resource_owner = 1 [
+        (validate.rules).string = {max_len: 200},
+        (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+            example: "\"69629023906488334\""
+        }
+    ];
+}
```
Comment (livio-a) [inline proto/zitadel/user/v2beta/query.proto:212]: please create a complete description (...always equals ?) and shouldn't the field be required / non-empty?
https://docs-git-user-v2-list-zitadel.vercel.app/docs/apis/resources/user_service_v3/user-service-list-users
image

File: proto/zitadel/user/v2beta/user_service.proto [hunk 3]
```diff
@@ -957,7 +953,6 @@ message GetUserByIDRequest {
       description: "User ID of the user you like to get."
     }
   ];
-  zitadel.object.v2beta.Organization organization = 2;
```
Comment (livio-a) [inline proto/zitadel/user/v2beta/user_service.proto:960]: i'd add a reserved also for the field name, just in case...

**Review by livio-a — COMMENTED** [review]
please also update the description with the breaking changes

File: proto/zitadel/user/v2beta/query.proto (hunk 1 above)
Comment (stebenz) [reply proto/zitadel/user/v2beta/query.proto:32]: True, will change it to organizationIdQuery, to keep it consistent

**Review by livio-a — APPROVED** [review]

**Pull request merged**